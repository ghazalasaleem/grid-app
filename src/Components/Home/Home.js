import React, { useEffect, useState } from 'react';
import './Home.scss';


const Home = () =>{

    const [time, setTime] = useState();

    useEffect(()=>{
        const date = new Date();
        const ticking = setInterval(()=>{setTime(date.getHours()+":"+date.getMinutes()+":"+date.getSeconds())}, 1000);

        return ()=> {clearInterval(ticking);}
    });
    return (
        <div className="home-container">
            <h2 className="msg">Hi Human!!</h2>
            <div className="timer">Time: <span>{time}</span></div>
        </div>);
}

export default Home;

{/* <button onClick={()=> setStop(!stop)}>Start/Stop</button> */}
