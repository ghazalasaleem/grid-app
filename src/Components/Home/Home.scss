.home-container{
    padding: 10px;

    // .msg{
    //     float: left;
    // }
    .timer{
        float: right;
        padding: 5px;
    }
    
}