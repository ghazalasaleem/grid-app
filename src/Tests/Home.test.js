import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../Components/Home/Home';

describe('Home Component', () =>{
  let home;
  
    beforeAll(() => {
      home = renderer.create(<Home/>);
    });
    afterAll(() => {
      home = null;
    });
     
    test('renders', () => {
      expect(home).toBeDefined();
      expect(home).toMatchSnapshot();
    });

    xtest('text displayed',() => {
      expect(home.value).toEqual('Simple Application to learn React');
    });
});

