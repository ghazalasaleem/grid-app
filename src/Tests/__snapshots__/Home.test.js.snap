// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders 1`] = `
<div
  className="home-container"
>
  <h2>
    Simple Application to learn React
  </h2>
</div>
`;
